### PR TITLE
Added note on YAML mappings as objects

### PR DESCRIPTION
**Pull request by dantleech**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.7 > 3.0 |

Depends on https://github.com/symfony/symfony-docs/pull/6582

### CODE REVIEW

File: components/yaml/introduction.rst [hunk 1]
```diff
@@ -148,6 +148,16 @@ error if something goes wrong by adding the filename to the message.
 
 .. _components-yaml-dump:
 
+Objects for Mappings
+....................
+
+Yaml mappings_ are basically associative arrays. You can instruct the parser to return
+mappings as objects (i.e. ``\stdClass`` instances)::
+
+    $object = $parser->parse('{"hello": "world"}', false, false, true);
```
Comment (xabbuh) [inline components/yaml/introduction.rst]: should make use of `Yaml::parse()`

File: components/yaml/introduction.rst [hunk 2]
```diff
@@ -219,6 +229,9 @@ representation to the inline one:
         foo: bar
         bar: baz
 
+
+
```
Comment (xabbuh) [inline components/yaml/introduction.rst]: these blank lines should be reverted

File: components/yaml/introduction.rst [hunk 3]
```diff
@@ -148,6 +148,16 @@ error if something goes wrong by adding the filename to the message.
 
 .. _components-yaml-dump:
 
+Objects for Mappings
+....................
+
+Yaml mappings_ are basically associative arrays. You can instruct the parser to return
```
Comment (xabbuh) [inline components/yaml/introduction.rst]: For interal links, we can use the `ref` role:

```rst
Yaml :ref:`mappings <yaml-format-collections>` are [...]
```

And we then need the label in the `/components/yaml/yaml_format.rst` file:

```rst
.. _yaml-format-collections:

Collections
-----------
```

**Comment by dantleech** [comment]
Updated.

**Comment by HeahDude** [comment]
closes 6524.

File: components/yaml/introduction.rst [hunk 4]
```diff
@@ -148,6 +148,21 @@ error if something goes wrong by adding the filename to the message.
...
+arrays. You can instruct the parser to return mappings as objects (i.e.
+``\stdClass`` instances)::
+
+    $object = Yaml::parse('{"foo": "bar"}', false, false, true);
+    echo get_class($object); // stdClass
+    echo $object->foo; // bar
+
+.. versionadded:: 2.7
+
+    This feature was introduced in Symfony 2.7
```
Comment (xabbuh) [inline components/yaml/introduction.rst]: I would move this just below the "Objects for Mappings" headline.

Comment (xabbuh) [inline components/yaml/introduction.rst]: And we can reword this a bit:

> Support for parsing mappings as objects was introduced in Symfony 2.7.

File: components/yaml/introduction.rst [hunk 5]
```diff
@@ -148,6 +148,21 @@ error if something goes wrong by adding the filename to the message.
...
+Yaml :ref:`mappings <yaml-format-collections>` are basically associative
+arrays. You can instruct the parser to return mappings as objects (i.e.
+``\stdClass`` instances)::
+
+    $object = Yaml::parse('{"foo": "bar"}', false, false, true);
+    echo get_class($object); // stdClass
+    echo $object->foo; // bar
+
+.. versionadded:: 2.7
+
```
Comment (WouterJ) [inline components/yaml/introduction.rst]: this empty line should be removed (versionadded directive is a strange one)

**Comment by dantleech** [comment]
Updated

File: components/yaml/introduction.rst [hunk 6]
```diff
@@ -148,6 +148,21 @@ error if something goes wrong by adding the filename to the message.
 
 .. _components-yaml-dump:
 
+Objects for Mappings
+....................
+
+.. versionadded:: 2.7
+
```
Comment (WouterJ) [inline components/yaml/introduction.rst]: after your pushes, it's this line that should be removed

File: components/yaml/introduction.rst [hunk 7]
```diff
@@ -148,6 +148,21 @@ error if something goes wrong by adding the filename to the message.
 
 .. _components-yaml-dump:
 
+Objects for Mappings
+....................
+
+.. versionadded:: 2.7
```
Comment (HeahDude) [inline components/yaml/introduction.rst]: 2.5?

File: components/yaml/introduction.rst [hunk 8]
```diff
@@ -148,6 +148,21 @@ error if something goes wrong by adding the filename to the message.
...
+
+    Support for parsing mappings as objects was introduced in Symfony 2.7.
+
+Yaml :ref:`mappings <yaml-format-collections>` are basically associative
+arrays. You can instruct the parser to return mappings as objects (i.e.
+``\stdClass`` instances)::
```
Comment (WouterJ) [inline components/yaml/introduction.rst]: "[...] by setting the fourth argument to true"

File: components/yaml/introduction.rst (hunk 6 above)
Comment (dantleech) [inline components/yaml/introduction.rst]: yeah i saw :)

**Comment by dantleech** [comment]
Updated ..

File: components/yaml/introduction.rst (hunk 7 above)
Comment (dantleech) [inline components/yaml/introduction.rst]: 2.6 :)

File: components/yaml/introduction.rst [hunk 9]
```diff
@@ -148,6 +148,20 @@ error if something goes wrong by adding the filename to the message.
 
 .. _components-yaml-dump:
 
+Objects for Mappings
+....................
+
+.. versionadded:: 2.6
+    Support for parsing mappings as objects was introduced in Symfony 2.6.
```
Comment (xabbuh) [inline components/yaml/introduction.rst]: It was added in 2.7.

**Comment by HeahDude** [comment]
👍

File: components/yaml/introduction.rst [hunk 10]
```diff
@@ -148,6 +148,20 @@ error if something goes wrong by adding the filename to the message.
...
+
+Yaml :ref:`mappings <yaml-format-collections>` are basically associative
+arrays. You can instruct the parser to return mappings as objects (i.e.
+``\stdClass`` instances) by setting the forth argument to ``true``::
```
Comment (HeahDude) [inline components/yaml/introduction.rst]: fourth

File: components/yaml/introduction.rst (hunk 9 above)
Comment (dantleech) [inline components/yaml/introduction.rst]: Hmm, it was added to `Parser` in 2.6 and and `Yaml::parse` in 2.7 :/

File: components/yaml/introduction.rst (hunk 7 above)
Comment (dantleech) [inline components/yaml/introduction.rst]: So it was added to `Parser#parse` in `2.6` but to `Yaml::parse` in 2.7. Kept the note at 2.7 as I guess `Yaml` is the main entry point.

Comment (xabbuh) [inline components/yaml/introduction.rst]: Ah indeed, this was a bit inconsistent. But as we should prefer the static API 2.7 is a good choice here.

File: components/yaml/introduction.rst (hunk 9 above)
Comment (xabbuh) [inline components/yaml/introduction.rst]: I think we should prefer the static API (see #6598) so using 2.7 here is the best idea imo.

**Comment by WouterJ** [comment]
:+1:

**Comment by xabbuh** [comment]
Thank you @dantleech!

**Pull request closed without merging**